### PR TITLE
Implementing function aliasByMetric

### PR DESCRIPTION
**Pull request by vaguiar**
**Describe your changes**
Implementing feature aliasByMetric

**Testing performed**
Unit tests
Integration tests

### CODE REVIEW

File: expr/func_aliasbymetric.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	return context
+}
+
+func (s *FuncAliasByMetric) Exec(dataMap DataMap) ([]models.Series, error) {
+	series, err := s.in.Exec(dataMap)
+	if err != nil {
+		return nil, err
+	}
+	for i, serie := range series {
+		n := extractMetric(serie.Target)
```
Comment (Dieterbe) [inline expr/func_aliasbymetric.go:31]: This is not what aliasByMetric does.
aliasByMetric takes the "base" of the metric as the alias (the last word)

e.g. `foo.bar.baz` -> `baz`
`perSecond(foo.bar.baz)` -> `baz`

extractMetric is used to extract full metric names.
so I think you should just keep `n := extractMetric(serie.Target)`, then find the last occurence of `.` within `n` and keep the string after that last `.`

Comment (vaguiar) [reply expr/func_aliasbymetric.go:31]: Thanks for the clarification. With the graphite implementation, the pattern we were seeing is if the metric contains tags with a period in them like `foo.bar.baz;tag1=one.blah;tag2=twoblah` what would be returned is `blah;tag2=twoblah` so everything after the last period is encountered. That's wrong and needs to be fixed in this implementation, correct?

Comment (replay) [reply expr/func_aliasbymetric.go:31]: yes, the behavior which you describe definitely sounds like a bug and isn't intended

Comment (vaguiar) [reply expr/func_aliasbymetric.go:33]: Roger that. I've made the required changes of extracting only the base name.
Now, the expected behavior is:
- `perSecond(foo.bar.baz)` -> `baz`
- `perSecond(foo.bar.baz;tag1=one.blah;tag2=twoblah)` -> `baz`
Please let me know if this is correct.

I'll re-request a review after testing it throughly on my end.

Comment (Dieterbe) [reply expr/func_aliasbymetric.go:33]: @vaguiar I recommend you use a graphite server to test against and see what the behavior is. (notwithstanding the bug we highlighted). When I test something like `aliasByMetric(perSecond(foo.bar.baz;tag1=one.blah;tag2=twoblah))` or just `aliasByMetric(foo.bar.baz;tag1=one.blah;tag2=twoblah)`, the output is `baz;tag1=one.blah;tag2=twoblah`

Comment (vaguiar) [reply expr/func_aliasbymetric.go]: Noted. I ran tests to compare the graphite output with that of metrictank. Thanks for the patient clarifications and apologies for the back and forth.

**Pull request merged**